### ISSUE TITLE
support for including local egg/wheel to pex

### ISSUE THREAD

**Issue by gunjan075** [pantsbuild/pex#2046]
requirements.txt
===============
pyspark
findspark
dynaconf
setuptools
wheel
**-f dist/abc.egg**

pex -r requirements.txt  -o pyspark_pex_env.pex -v

pyspark_pex_env.pex should have abc also
import abc is successfull

**Comment by jsirois** [comment]
Pex does support local wheel dependencies but it does not support eggs at all. There is no intention of adding support for them since they are, by now, decidedly non-standard: https://packaging.python.org/en/latest/discussions/wheel-vs-egg/

What software are you using that still uses the egg format?

**Comment by gunjan075** [comment]
I am using spark. Spark just hates wheel file.

So if i replace. egg with wheel will it work?? 

-f dist/abc.whl

**Comment by jsirois** [comment]
@gunjan075 yes, it will work. That said, you're not using -f (--find-links) correctly. From the Pex help:
```
  -f PATH/URL, --find-links PATH/URL, --repo PATH/URL
                        Additional repository path (directory or URL) to look
                        for requirements. (default: None)
```

And the corresponding Pip help:
```
  -f, --find-links <url>      If a URL or path to an html file, then parse for links to archives such as sdist (.tar.gz) or wheel (.whl) files. If a local path or file:// URL that's a directory, then look for
                              archives in the directory listing. Links to VCS project URLs are not supported.
```

In short, when using a find-links repo that is a local, it should be a directory containing distributions and not the path to the distribution itself; so just `-f dist/` in your case. But then you need to specify the requirement name too. In total, it would look like:
```
pyspark
findspark
dynaconf
setuptools
wheel
abc
-f dist/
```

And that is true when using Pip or Pex.

What you probably really want though is to drop the -f. Both Pex and Pip can work with local wheels as requirements:
```
pyspark
findspark
dynaconf
setuptools
wheel
dist/abc.whl
```

Guessing a bit though, a `dist/` directory generally indicates you pre-built the dist before calling Pex. Pex knows how to build local projects too; so you likely could just say:
```
pyspark
findspark
dynaconf
setuptools
wheel

# The following `.` is a requirement on the project in the current
# directory. Pex knows how to build local projects and will turn
# this into a wheel for you behind the scenes.
.
```

**Comment by jsirois** [comment]
@gunjan075 let me know if any of the above suggestions work for you and I'll close this question as answered.

**Comment by gunjan075** [comment]
thank you that was helpfull
